GVFS/GVFS/CommandLine/ConfigVerb.cs

```diff
@@ -31,30 +46,95 @@ protected override string VerbName
 
         public override void Execute()
         {
-            this.localConfig = new LocalGVFSConfig();
+            if (GVFSPlatform.Instance.IsUnderConstruction)
+            {
+                this.ReportErrorAndExit("`gvfs config` is not yet implemented on this operating system.");
+            }
 
+            this.localConfig = new LocalGVFSConfig();
             string error = null;
-            bool isRead = string.IsNullOrEmpty(this.Value);
 
-            if (isRead)
+            if (this.IsMutuallyExclusiveOptionsSet(out error))
             {
-                string value = null;
-                if (this.localConfig.TryGetConfig(this.Key, out value, out error, tracer: null))
+                this.ReportErrorAndExit(error);
+            }
+
+            if (this.List)
+            {
+                Dictionary<string, string> allSettings;
+                if (!this.localConfig.TryGetAllConfig(out allSettings, out error))
                 {
-                    Console.WriteLine(value);
+                    this.ReportErrorAndExit(error);
                 }
-                else
+
+                const string ConfigOutputFormat = "{0}={1}";
+                foreach (KeyValuePair<string, string> setting in allSettings)
                 {
-                    this.ReportErrorAndExit(error);
+                    Console.WriteLine(ConfigOutputFormat, setting.Key, setting.Value);
                 }
             }
-            else
+            else if (!string.IsNullOrEmpty(this.KeyToDelete))
             {
-                if (!this.localConfig.TrySetConfig(this.Key, this.Value, out error, tracer: null))
+                if (!this.localConfig.TryRemoveConfig(this.KeyToDelete, out error))
                 {
                     this.ReportErrorAndExit(error);
                 }
             }
+            else if (!string.IsNullOrEmpty(this.Key))
+            {
+                bool valueSpecified = !string.IsNullOrEmpty(this.Value);

```

Comment:
git allows you to set a key to blank by running `git config key ""`. 
`git config -l` will then show
`key=`

Do we want to match that behavior?

Reply:
I don't think it's a goal to match git's behavior exactly. Where I've referred to `git config`, it was just to make sure we're taking a reasonably consistent approach. But we have no need for blank config values so I don't think we want to match this.